sns/src/main/scala/akka/stream/alpakka/sns/SnsPublishSinkStage.scala

```diff
@@ -0,0 +1,77 @@
+package akka.stream.alpakka.sns
+
+import akka.Done
+import akka.stream.stage._
+import akka.stream.{Attributes, Inlet, SinkShape}
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.sns.AmazonSNSAsync
+import com.amazonaws.services.sns.model.{PublishRequest, PublishResult}
+
+import scala.concurrent.{Future, Promise}
+
+final class SnsPublishSinkStage(topicArn: String, snsClient: AmazonSNSAsync)
```

Comment:
Is the `AmazonSNSAsync` thread safe? A stage can be materialized any number of times and all of them will share the same `snsClient`.  If it cannot be shared a factory lambda is better (`() => AmazonSNSAsync`), that way every materialization can have its own client (which it will be responsible to manage/close etc in sync with the stage lifecycle).

Another option might be to take a `AmazonSNSAsyncClientBuilder` and `.build()` that each materialization (if the builder is immutable that is).